tap_appstore/streams.py

```diff
@@ -1,72 +1,292 @@
+        super().__init__(*args, **kwargs)
+        self.int_fields = ['consecutive_paid_periods', 'app_apple_id', 'subscription_apple_id', 'subscription_group_id',
+                           'days_canceled', 'quantity']
+
+    def download_data(self, start_date, api):
+        filters = {
+            'frequency': 'DAILY',
+            'reportType': 'SUBSCRIPTION',
+            'reportSubType': 'SUMMARY',
+            'reportDate': self.get_report_date(start_date),
+            'version': '1_3',
+            'vendorNumber': self.config['vendor_number']
+        }
+        return api.download_sales_and_trends_reports(filters=filters)
 
-class GroupsStream(AppStoreStream):
-    """Define custom stream."""
 
-    name = "groups"
-    path = "/groups"
-    primary_keys: t.ClassVar[list[str]] = ["id"]
-    replication_key = "modified"
+class FinancialReportStream(client.AppStoreStream):
+    name = "financial_reports"
     schema = th.PropertiesList(
-        th.Property("name", th.StringType),
-        th.Property("id", th.StringType),
-        th.Property("modified", th.DateTimeType),
+        th.Property("_line_id", th.IntegerType),
+        th.Property("_time_extracted", th.StringType),
+        th.Property("_api_report_date", th.StringType),
+        th.Property("start_date", th.DateTimeType),
+        th.Property("end_date", th.DateTimeType),
+        th.Property("vendor_identifier", th.StringType),
+        th.Property("quantity", th.IntegerType),
+        th.Property("partner_share", th.NumberType),
+        th.Property("extended_partner_share", th.NumberType),
+        th.Property("partner_share_currency", th.StringType),
+        th.Property("sales_or_return", th.StringType),
+        th.Property("apple_identifier", th.StringType),
+        th.Property("title", th.StringType),
+        th.Property("product_type_identifier", th.StringType),
+        th.Property("country_of_sale", th.StringType),
+        th.Property("pre_order_flag", th.StringType),
+        th.Property("promo_code", th.StringType),
+        th.Property("customer_price", th.NumberType),
+        th.Property("customer_currency", th.StringType)
     ).to_dict()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.float_fields = ['customer_price', 'partner_share', 'extended_partner_share']
+        self.int_fields = ['quantity']
+        self.date_fields = {
+            'start_date': '%m/%d/%Y',
+            'end_date': '%m/%d/%Y',
+        }
+
+    def download_data(self, start_date, api):
+        filters = {'vendorNumber': self.config['vendor_number'],
+                   'regionCode': 'US',
+                   'reportType': 'FINANCIAL',
+                   'reportDate': self.get_report_date(start_date),
+                   }
+        return api.download_finance_reports(filters=filters)
+
+    def increment_date(self, date):
+        """Increment date by one month."""
+        return date + relativedelta(months=1)
```

Comment:
Let's use a class variable with the delta so we don't need to override the method in this Stream.